### PR TITLE
feat: generate a sitemap for Next.js version

### PR DESCRIPTION
**Pull request by suevalov**
# Purpose of PR

The sitemap will be used by crawlers and Algolia search crawlers specifically.

### CODE REVIEW

File: packages/website/scripts/generate-sitemap.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import prettier from 'prettier';
+import { getMdxPaths } from '../utils/content';
+
+dotenv.config();
+
+async function generate() {
+  // todo: replace with f36.contentful.com when we release v4
+  const baseUrl = 'https://v4-forma-36.vercel.app';
+
+  const staticSlugs = [''];
```
Comment (bgutsol) [inline packages/website/scripts/generate-sitemap.ts:12]: I guess this is for other pages that are not mdx based. And we will have to remember to include them here, right?

File: packages/website/scripts/generate-sitemap.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import dotenv from 'dotenv';
+import { writeFileSync } from 'fs';
+import prettier from 'prettier';
+import { getMdxPaths } from '../utils/content';
+
+dotenv.config();
```
[the remark this answers is not in the document]
Comment (suevalov) [reply packages/website/scripts/generate-sitemap.ts:6]: I was planning to use the env variable in this file. Maybe there's a way to get a deployment URL here.

File: packages/website/public/robots.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# *
+User-agent: *
+Allow: /
+
+# Host
+Host: https://v4-forma-36.vercel.app
+
+# Sitemaps
+Sitemap: https://v4-forma-36.vercel.app/sitemap.xml
```
[the remark this answers is not in the document]
Comment (suevalov) [reply packages/website/public/robots.txt:9]: I'll try to generate this file on a fly and read VERCEL_URL from env, let's see if it works

File: packages/website/scripts/generate-sitemap.ts (hunk 1 above)
Comment (suevalov) [reply packages/website/scripts/generate-sitemap.ts:12]: yes, I'm going to add a comment

Comment (suevalov) [reply packages/website/scripts/generate-sitemap.ts]: added a comment

**Review by bgutsol — APPROVED** [review]

**Pull request merged**